go/types/sequence_cursor.go

```diff
@@ -109,26 +243,49 @@ func (cur *sequenceCursor) retreatMaybeAllowBeforeStart(allowBeforeStart bool) b
 	return false
 }
 
+/*
+ * clone creates a copy of the cursor
+ *
+ * The clone does not inherit read-ahead behavior
+ */
 func (cur *sequenceCursor) clone() *sequenceCursor {
 	var parent *sequenceCursor
 	if cur.parent != nil {
 		parent = cur.parent.clone()
 	}
-	return &sequenceCursor{parent, cur.seq, cur.idx}
+	return newSequenceCursor(parent, cur.seq, cur.idx)
 }
 
 type cursorIterCallback func(item interface{}) bool
 
+/*
+ * iter iterates forward from the current position
+ */
 func (cur *sequenceCursor) iter(cb cursorIterCallback) {
+	cur.enableReadAhead() // ensure read-ahead is enabled to
 	for cur.valid() && !cb(cur.getItem(cur.idx)) {
 		cur.advance()
 	}
 }
 
+/*
+ * newCursorAtIndex creates a new cursor over seq positioned at idx.
+ *
+ * Implemented by searching down the tree to the leaf sequence containing idx. Each
+ * sequence cursor includes a back pointer to its parent so that it can follow the path
+ * to the next leaf chunk when the cursor exhausts the entries in the current chunk.
+ */
 func newCursorAtIndex(seq sequence, idx uint64) *sequenceCursor {
 	var cur *sequenceCursor
 	for {
 		cur = newSequenceCursor(cur, seq, 0)
+		// Assume cursor starting at 0 will be moving forward and benefit from read-ahead
+		// TODO: Inferring that a 0 idx means the cursor will be moving forward is
```

Comment:
The only time retreat is even used is priming the sequence chunker.

If that doesn't mean anything to you: when modifying a prolly-tree, we use a sequence chunker, which needs to
(a) set the rolling hash to the correct value (requires retreating the hash window) and
(b) re-start from the beginning of the chunk at the modification point (also requires retreating).

Straight after doing that, we advance back to where we started. Furthermore, after that PR I just sent you, we later use that same cursor to read to the end of the chunk.

We should read that ahead.

Point being: just always read ahead.